ouroboros-network/src/Ouroboros/Network/PeerSelection/Churn.hs

```diff
@@ -44,30 +59,52 @@ peerChurnGovernor :: forall m peeraddr.
                   -> StrictTVar m ChurnMode
                   -> StdGen
                   -> STM m FetchMode
-                  -> PeerSelectionTargets
+                  -> TargetsSelector
                   -> StrictTVar m PeerSelectionTargets
                   -> STM m UseBootstrapPeers
+                  -> StrictTMVar m LedgerStateJudgement
+                  -- ^ Peer selection govnr chooses the ledger state that churn
+                  -- govnr should use for setting targets to keep everything in sync
+                  -> Bool
                   -> m Void
 peerChurnGovernor tracer deadlineChurnInterval bulkChurnInterval psOverallTimeout
-                  _metrics churnModeVar inRng getFetchMode base peerSelectionVar
-                  getUseBootstrapPeers = do
+                  _metrics churnModeVar inRng getFetchMode targetsSelector peerSelectionVar
+                  getUseBootstrapPeers mutexPeerSelection useGenesis = do
```

Comment:
a.d. `mutexPeereSelection`: it would be good to use the same name as on the `PeerSelection` side.

Reply:
so there are three different names for this variable actually. When reading P2P.hs on line 812, it is called churnMutexPeerSelection to suggest to the reader that this mutually excludes those two components, since at that point we are outside both. In Peer selection governor I called it churnMutex, to suggest that the mutual exclusion comes from churn, and in churn governor it is called mutexPeerSelection to suggest that the mutual exclusion may come from there. So there was some logic behind the madness, and it might have been a bad idea :( but it wasn't just random.

Reply:
Yes, I got that; but it hides that this is the same mutex.

Reply:
So I've written a version without using a mutex, it seems to pass the tests, but I'll leave this one as-is for now since I've ran the node with it for testing the peer snapshot and it works. The version w/o the mutex will be a separate PR, probably along with integrating churn into the OG.